aries_cloudagent/resolver/base.py

```diff
@@ -75,16 +90,19 @@ def supports(self, did: Union[str, DID]) -> bool:
 
         return False
 
-    async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
+    async def resolve(self, profile: Profile, did: Union[str, DID]) -> ResolutionResult:
```

Comment:
Sorry, this is changing the requirements of this task a bit but it struck me that we might want to split things up into two different methods; i.e. `resolve` vs `resolve_with_metadata`. In this scenario, if the caller doesn't care about the resolution metadata, they can just resolve directly the document and use it without needing to unwrap. If they do care, then the extra unwrap step when they're ready to use the document is not an extraneous step. Thoughts?

Reply:
Hmmm I think it touches so many things. What do you think about a kwargs flag to build the metadata or not? something like `def resolve(self, profile: Profile, did: Union[str, DID], retrieve_metadata: bool = False)`